### PR TITLE
fix: just merge vault pnl points for smooth recent pnl plus long term views

### PR DESCRIPTION
**Pull request by tyleroooo**
Going to hold off merging until staging pnl points are fixed so I can see how this actually looks.

### CODE REVIEW

**Comment by tyleroooo** [comment]
seems fine

File: src/hooks/vaultsHooks.ts [hunk 1]
```diff
@@ -67,10 +68,18 @@ export const useLoadedVaultDetails = () => {
   const vaultDetailsResult = useQuery({
     queryKey: ['vaultDetails'],
     queryFn: async () => {
-      const callResult = await getMegavaultHistoricalPnl();
+      const toTyped = (res: any) =>
```
Comment (tinaszheng) [inline src/hooks/vaultsHooks.ts:71]: you can probs move this function definition out of this queryFn

File: src/hooks/vaultsHooks.ts [hunk 2]
```diff
@@ -67,10 +68,18 @@ export const useLoadedVaultDetails = () => {
...
+        VaultCalculator.getVaultHistoricalPnlResponse(safeStringifyForAbacusParsing(res));
+      const [dailyResult, hourlyResult] = await Promise.all([
+        getMegavaultHistoricalPnl().then(toTyped),
```
Comment (tinaszheng) [inline src/hooks/vaultsHooks.ts:74]: maybe just pass in the PnlTickInterval.day here to be explicit?

also why is it `PnlTickInterval.day` but `PnlTickInterval.HOUR` is all caps lmao

**Review by tinaszheng — APPROVED** [review]

File: src/hooks/useDydxClient.tsx [hunk 3]
```diff
@@ -169,14 +170,17 @@ const useDydxClientContext = () => {
...
-      return await indexerClient.vault.getMegavaultHistoricalPnl();
-    } catch (error) {
-      log('useDydxClient/getMegavaultHistoricalPnl', error);
-      return undefined;
-    }
-  }, [indexerClient.vault]);
+  const getMegavaultHistoricalPnl = useCallback(
+    async (resolution: PnlTickInterval = PnlTickInterval.day) => {
+      try {
+        return await indexerClient.vault.getMegavaultHistoricalPnl(resolution);
```
Comment (tinaszheng) [inline src/hooks/useDydxClient.tsx:176]: why not just wrap this with `  VaultCalculator.getVaultHistoricalPnlResponse(safeStringifyForAbacusParsing(res));` here

File: src/hooks/vaultsHooks.ts (hunk 2 above)
Comment (tyleroooo) [reply src/hooks/vaultsHooks.ts:74]: lol I was wondering that too

**Pull request merged**